README.md

```diff
@@ -8,16 +8,207 @@ Local.
 For more information, see [Getting started with Cloudant NoSQL DB](https://www.ng.bluemix.net/docs/services/Cloudant/index.html)
 
 The `loopback-connector-cloudant` module is the Cloudant connector for the Loopback framework.
+<!-- TOC -->
+
+- [loopback-connector-cloudant](#loopback-connector-cloudant)
+- [Getting Started](#getting-started)
+    - [Design](#design)
+        - [Careful partial update](#careful-partial-update)
+        - [Careful Frequent Modification](#careful-frequent-modification)
+        - [Conflict Control](#conflict-control)
+    - [Model](#model)
+        - [Map between model and document](#map-between-model-and-document)
+        - [Model-specific configuration](#model-specific-configuration)
+        - [_rev property](#_rev-property)
+- [Setup Cloudant Instance](#setup-cloudant-instance)
+- [Installation](#installation)
+- [Configuration](#configuration)
+    - [Generate Datasource](#generate-datasource)
+    - [Datasource Config(`url` vs `username` & `password`)](#datasource-configurl-vs-username--password)
+    - [Requests Plugin](#requests-plugin)
+    - [Example Usage](#example-usage)
+- [CRUD](#crud)
+    - [Update](#update)
+- [Migration](#migration)
+    - [autoupdate vs automigrate](#autoupdate-vs-automigrate)
+    - [isActual](#isactual)
+    - [Example Code](#example-code)
+- [Discovery](#discovery)
+- [Query](#query)
+- [Testing](#testing)
+- [More Info](#more-info)
+- [Feature backlog](#feature-backlog)
+    - [View](#view)
+    - [Index](#index)
+
+<!-- /TOC -->
+
+# Getting Started
+
+## Design
+Loopback tries best to fit its model to a specific database's design, while limited by the nature of database, it's not always possible to support all Loopback features perfectly, and user would like to be aware of some key features about Cloudant before they start to design a Cloudant model.
+
+### Careful partial update
+
+*Cloudant does not support the idea of updating a document. All "updates" on a document are _destructive_ replacements.*
 
-## Key features
+For example:
+
+```
+// original document
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "1",
+  "prop2": "2",
+}
+
+// data to be updated
+ds.updateOrCreate('User', {
+  prop1: 'updated1',
+}, function (err, res) {});
+
+// document after update
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "updated1",
+}
+```
+
+*Please note how property `prop2` was completely dropped upon update.*
+
+Workaround: Do not pass partial values for the data object to be updated. If there are properties that are not being updated, please include the old value to keep data persistent.
+
+We have some discussion on update methods, the issue link can be found in [Feature Backlog](https://github.com/strongloop/loopback-connector-cloudant##feature-backlog) section
+
+### Careful Frequent Modification
```

Comment:
I think having just `Frequent Modification` here is enough, but am okay with this as well.